### PR TITLE
Use a factory for performance counters.

### PR DESCRIPTION
**Pull request by mateoatr**
**+** Initialize performance counters using a factory.
**+** Make a distinction between the name of the counter and its name in the json.
**+** StyleCop edits.
**+** Rename files so that they match the name of the class.

### CODE REVIEW

File: AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs [hunk 1]
```diff
@@ -8,28 +8,36 @@
...
+        private CacheHelper()
+        {
+        }
 
         /// <summary>
-        /// Return the only instance of CacheHelper.
+        /// Gets the only instance of CacheHelper.
         /// </summary>
-        public static CacheHelper Instance
+        public static CacheHelper GetInstance
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs]: Property name should not have a verb into it.

File: AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs [hunk 2]
```diff
@@ -68,22 +76,22 @@ public int PerformanceCounterValue(string performanceCounterName, string json)
         /// <returns>value from cache</returns>
         public int GetCounterValue(string name)
         {
-            const string jsonKey = "json";
-            if (!CacheHelper.Instance.IsInCache(jsonKey))
+            const string JonKey = "json";
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs]: strange renaming

File: AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CounterFactory.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                                "readIoBytes", 
+                                "readIoBytes"),
+                            new PerformanceCounterFromJsonGauge(
+                                "writeIoBytes", 
+                                "writeIoBytes"),
+                            new PerformanceCounterFromJsonGauge(
+                                "otherIoBytes", 
+                                "otherIoBytes")));
+                default:
+                    throw new InvalidEnumArgumentException();
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CounterFactory.cs]: should be just ArgumentException. No enum here

File: AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CounterFactory.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    return new PerformanceCounterFromJsonGauge(
+                        @"\ASP.NET Applications(??APP_W3SVC_PROC??)\Request Execution Time",
+                        "appRequestExecTime");
+                case @"\Process(??APP_WIN32_PROC??)\Private Bytes":
+                    return new PerformanceCounterFromJsonGauge(
+                        @"\Process(??APP_WIN32_PROC??)\Private Bytes",
+                        "privateBytes");
+                case @"\ASP.NET Applications(??APP_W3SVC_PROC??)\Requests In Application Queue":
+                    return new PerformanceCounterFromJsonGauge(
+                        @"\ASP.NET Applications(??APP_W3SVC_PROC??)\Requests In Application Queue",
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CounterFactory.cs]: instead of repeating the string - just pass `counterName`

**Comment by SergeyKanzhelev** [comment]
Almost good from the first attempt

File: AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs (hunk 2 above)
Comment (mateoatr) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs]: StyleCop recommends using Pascal casing for constants.

Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs]: make it class-level constant then please

Comment (dnduffy) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/Implementation/CacheHelper.cs]: Correct the spelling of "Json" (currently it says "Jon" which is a person's name).

**Pull request merged**